### PR TITLE
Add h2 enhancer

### PR DESCRIPTION
**Pull request by abeddow91**
## What does this change?
Adds a new enhancer to enhance h2 tags with an id when they are type `SubheadingBlockElement`.

This was previously done within the `enhance-interactive-contents-elements.ts` enhancer and only for interactive content block elements.

It also adds tests for the h2 enhancer. 

## Why?
We would like to have ids on h2 elements so that we can navigate to them more easily . 

By moving the id enhancement into its own enhancer which runs before `enhance-interactive-contents-elements.ts`, we retain the existing functionality but open up this enhancement beyond just Interactive block elements.

### CODE REVIEW

File: dotcom-rendering/src/model/enhanceBlocks.ts [hunk 1]
```diff
@@ -85,6 +91,7 @@ export const enhanceBlocks = (
 ): Block[] => {
 	return new BlockEnhancer(blocks, format, promotedNewsletter)
 		.enhanceDividers()
+		.enhanceH2()
```
Comment (oliverlloyd) [inline dotcom-rendering/src/model/enhanceBlocks.ts:94]: Maybe `enhanceH2s`?

Comment (abeddow91) [reply dotcom-rendering/src/model/enhanceBlocks.ts:94]: 👍

File: dotcom-rendering/src/model/enhance-H2s.test.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+						elementId: 'mockId',
+						html: "<h2 id='mockId'>I should get an id.</h2>",
+					},
+				],
+			},
+		];
+
+		expect(enhanceH2s(input)).toEqual(expectedOutput);
+	});
+	it('does not set an id when it is an h2 not of type TextBlockElement', () => {
```
Comment (oliverlloyd) [inline dotcom-rendering/src/model/enhance-H2s.test.ts:34]:
```suggestion
	it('does not set an id for h2s of type TextBlockElement', () => {
```

**Review by oliverlloyd — APPROVED** [review]
👌

File: dotcom-rendering/src/model/enhance-interactive-contents-elements.ts [hunk 3]
```diff
@@ -22,37 +22,24 @@ const enhance = (elements: CAPIElement[]): CAPIElement[] => {
 
 	if (hasInteractiveContentsBlockElement) {
 		// We want to record all `SubheadingBlockElement` to construct the interactive content block
-		const subheadingLinks: SubheadingBlockElement[] = [];
-		const withUpdatedSubheadings: CAPIElement[] = elements.map(
-			(thisElement) => {
-				if (
-					thisElement._type ===
+		const subheadingLinks = elements.filter(
+			(element): element is SubheadingBlockElement => {
```
Comment (mxdvl) [inline dotcom-rendering/src/model/enhance-interactive-contents-elements.ts:26]: Superb 👌

**Pull request merged**